driver-core/src/test/unit/com/mongodb/AuthConnectionStringTest.java

```diff
@@ -104,9 +99,21 @@ private void testValidUris() {
             }
 
             assertMechanism("credential.mechanism", credential.getMechanism());
+        } else {
+            if (!getExpectedValue("credential").equals(BsonNull.VALUE)) {
+                fail(String.format("Connection string '%s' should not produce credentials", input));
+            }
         }
     }
 
+    @Nullable
+    private MongoCredential getMongoCredential() {
+        ConnectionString connectionString;
```

Comment:
`return new ConnectionString(input).getCredential()` is simpler

Reply:
Looks like an extraction artifact. I have:
```
        if (credential != null) {
            BsonArray callbacks = (BsonArray) getExpectedValue("callback");
        ...
```
immediately after the `credential =` line. I could condense, but then I would have to expand it later. Do you have a preference?

Reply:
Small enough that it's fine to just expand it later.

Reply:
Did you end up simplifying this as I requested?

Reply:
Done now